src/sentry/migrations/0515_slugify_invalid_monitors.py

```diff
@@ -0,0 +1,106 @@
+# Generated by Django 2.2.28 on 2023-07-07 15:51
+from django.db import IntegrityError, migrations
+from django.utils.text import slugify
+
+from sentry.constants import ObjectStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def schedule(cls, instance, days=30):
+    from datetime import timedelta
+
+    from django.utils import timezone
+
+    model = type(instance)
+    model_name = model.__name__
+    cls.objects.update_or_create(
+        app_label=instance._meta.app_label,
+        model_name=model_name,
+        object_id=instance.pk,
+        defaults={
+            "actor_id": None,
+            "data": {},
+            "date_scheduled": timezone.now() + timedelta(days=days, hours=0),
+        },
+    )
+
+
+def delete_rules(apps, schema_editor):
+    Rule = apps.get_model("sentry", "Rule")
+    RegionScheduledDeletion = apps.get_model("sentry", "RegionScheduledDeletion")
+    for rule in RangeQuerySetWrapperWithProgressBar(Rule.objects.all()):
+        if rule.status in (ObjectStatus.PENDING_DELETION, ObjectStatus.DISABLED):
+            schedule(RegionScheduledDeletion, rule, days=0)
+
+
+def migrate_monitor_slugs(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+    Rule = apps.get_model("sentry", "Rule")
+
+    MAX_SLUG_LENGTH = 50
+    monitors_to_clean_up = []
+
+    for monitor in RangeQuerySetWrapperWithProgressBar(Monitor.objects.all()):
+        monitor_slug = monitor.slug
+        slugified = slugify(monitor_slug)[:MAX_SLUG_LENGTH].strip("-")
+
+        # Nothing to migrate if the monitor already has a valid slug
+        if monitor_slug == slugified:
+            continue
+
+        try:
+            monitor.slug = slugified
+            monitor.save()
+        except IntegrityError:
+            # If there is a collision, delete the old monitor as the new one is receiving all check-ins
+            monitors_to_clean_up.append(monitor.id)
+
+    for delete_monitor in Monitor.objects.filter(id__in=monitors_to_clean_up):
+        alert_rule_id = delete_monitor.config.get("alert_rule_id")
+        if alert_rule_id:
+            rule = (
+                Rule.objects.filter(
+                    project_id=delete_monitor.project_id,
+                    id=alert_rule_id,
+                )
+                .exclude(
+                    status__in=[
+                        ObjectStatus.PENDING_DELETION,
+                        ObjectStatus.DELETION_IN_PROGRESS,
+                    ]
+                )
+                .first()
+            )
+            if rule:
+                rule.status = ObjectStatus.PENDING_DELETION
+                rule.save()
+
+        delete_monitor.status = ObjectStatus.PENDING_DELETION
+        delete_monitor.save()
```

Comment:
You still need to call schedule here

Reply:
yeah had a bunch of local changes I didn't realize got overwritten with a bad merge/update migration script